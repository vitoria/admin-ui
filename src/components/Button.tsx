import React, { forwardRef, Ref } from 'react'
import { Button as WAIButton, ButtonProps as WAIProps } from 'reakit'
import { Button as TAButton, ButtonProps as TAProps } from 'theme-ui'

type Props = WAIProps & TAProps

function Button(props: Props, ref: Ref<HTMLButtonElement>) {
  return <WAIButton {...props} ref={ref} as={TAButton} />
}

export default forwardRef(Button)
