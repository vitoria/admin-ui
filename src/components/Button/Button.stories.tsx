import React from 'react'

import Provider from '../ThemeProvider'
import Button from '.'

export default {
  title: 'Button',
  component: Button,
}

export const Themes = () => {
  return (
    <Provider>
      <Button bg="primary">Primary Button</Button>
      <Button bg="secondary">Seconday Button</Button>
    </Provider>
  )
}
