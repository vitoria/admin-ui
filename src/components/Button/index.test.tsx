import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'

import Button from '.'

describe('Button tests', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Button>Example</Button>)

    expect(asFragment()).toMatchSnapshot()
  })
})
