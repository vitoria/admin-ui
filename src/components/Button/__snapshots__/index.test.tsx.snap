// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="css-vhv7r1"
    type="button"
  >
    Example
  </button>
</DocumentFragment>
`;
